snuba/subscriptions/scheduler.py

```diff
@@ -321,6 +328,36 @@ def __reset_builder(self) -> None:
             # We are transitioning between jittered and immediate mode. We must use the delegate builder.
             self.__builder = self.__delegate_builder
 
+    def __filter_subscriptions(self) -> List[Subscription]:
+
+        # only generic metrics storage sets are currently sliced
+        if (
+            self.__entity_key == EntityKey.GENERIC_METRICS_SETS
+            or self.__entity_key == EntityKey.GENERIC_METRICS_DISTRIBUTIONS
+        ):
+
+            filtered_subscriptions: List[Subscription] = []
+            for subscription in self.__subscriptions:
+                # get the metadata and org_id from the Subscription
+                sub_data = subscription.data
+                sub_metadata = sub_data.metadata
+                org_id = sub_metadata["organization"]
```

Comment:
should we do something if `organization` is None? maybe we can emit a metric and skip the current subscription?

Reply:
good point, done